### ISSUE TITLE
Failed to send message: hipchat-message

### ISSUE THREAD

**Issue by ajoaugustine** [saltstack/salt#42438]
### Description of Issue/Question
Issue: Not able to pass the keyword color for  'hipchat.send_message'. Status shows failed. However the message is getting posted at the hip chat.
Pasting the error message:
[ERROR   ] Failed to send message: hipchat-message
local:
----------
          ID: hipchat-message
    Function: hipchat.send_message
      Result: False
     Comment: Failed to send message: hipchat-message
     Started: 19:11:07.306542
    Duration: 245.703 ms
     Changes:
    Warnings: 'color' is an invalid keyword argument for 'hipchat.send_message'.
              If you were trying to pass additional data to be used in a
              template context, please populate 'context' with 'key: value'
              pairs. Your approach will work until Salt Oxygen is out. Please
              update your state files.

Summary for local
------------
Succeeded: 1
Failed:    1
Warnings:  1
------------
Total states run:     2
Total run time: 748.277 ms

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)
sls file:
  hipchat.send_message:
    - room_id: 40305
    - from_name: SuperAdmin
    - message: 'This state was executed successfully.'
    - api_url: https://api.hipchat.com
    - api_key: uDMxaRzJNyIuIOIobUhM7BgHCgOtKf
    - api_version: v2
    - color: 'red'
    - notify: True

### Steps to Reproduce Issue
execute the state with a valid api key and room ID.

To fix the 'color' related error, you need to replace color with 'message_color' in the sls file and then modify the documentation:
https://docs.saltstack.com/en/latest/ref/states/all/salt.states.hipchat.html

or update /usr/lib/python2.7/dist-packages/salt/states/hipchat.py

### Versions Report
Salt Version:
           Salt: 2017.7.0

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Jul  1 2016, 15:12:24)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-38-generic
         system: Linux
        version: Ubuntu 16.04 xenial

**Comment by gtmanfred** [comment]
@anitakrueger or @mirceaulinic could one of yall look at this?

Unfortunately we do not have a hipchat setup to test this integration, If you have time to submit a PR to fix this, we would greatly appreciate it.  We would be able to fix the documentation.

Thanks,
Daniel

**Comment by mirceaulinic** [comment]
@gtmanfred @ajoaugustine at a glance, I understand that's just a documentation issue: the state calles the `hipchat.send_message` execution function with `color=message_color` (https://github.com/saltstack/salt/blob/develop/salt/states/hipchat.py#L132), while the function has a valid `color` arg: https://github.com/saltstack/salt/blob/develop/salt/modules/hipchat.py#L311. Or  do I miss anything here?

I suppose the following SLS should work:

```yaml
hipchat.send_message:
  - room_id: 40305
  - from_name: SuperAdmin
  - message: 'This state was executed successfully.'
  - api_url: https://api.hipchat.com
  - api_key: uDMxaRzJNyIuIOIobUhM7BgHCgOtKf
  - api_version: v2
  - message: 'red'
  - notify: True
```

Also: if you're using the public hipchat API (https://api.hipchat.com), you don't need to explicitly configure the `api_url` field.

**Comment by anitakrueger** [comment]
Looks like a documentation issue. The example in https://docs.saltstack.com/en/latest/ref/states/all/salt.states.hipchat.html just says "color" instead of "message_color" and so does the description of the parameters. The function definition for send_messages shows the correct parameter name though.

**Comment by danlsgiga** [comment]
This is not only a doc issue. The Hipchat module is considering an HTTP 204 (No Content) response as failed, when their specs says that is an expected response.

https://www.hipchat.com/docs/apiv2/method/send_room_notification

https://github.com/saltstack/salt/blob/a0716643e4f1c7dbbf97235b6c4e02920672b13f/salt/modules/hipchat.py#L179-L180

I've just tested setting the L180 to return True and it works as expected.

More on the doc issue, I personally lost almost 1h trying to troubleshoot why color was not a valid parameter when the docs examples use that.

From my perspective this is such a low hang fruit fix. Hoping this can make into the 2017.7.5 release.

**Comment by gtmanfred** [comment]
If you could submit a fix to the 2017.7 branch it could make it into 2017.7.5

Thanks,
Daniel

**Comment by danlsgiga** [comment]
No worries. I will submit a PR on monday!

**Comment by gtmanfred** [comment]
That would be awesome! thanks!
Daniel

**Comment by danlsgiga** [comment]
Hey @gtmanfred, I've just submitted the fix #46333 for this issue + a bonus fix on the `message_color` and `color` doc mismatch that fits into the 2017 release.
If that's not asking too much, I'd love to have that into 2017.7.5.
Thanks!!

**Comment by gtmanfred** [comment]
That looks great.

Since it is on the 2017.7 branch, and we haven't forked a release branch for 2017.7.5, this should get merged before that happens, and will be in the 2017.7.5 release.

I did have one change that needs to be made.  We do not like making breaking changes to code in minor releases, even though the docs are incorrect, they docs should be changed instead of changing the code.

Thanks,
Daniel

**Comment by danlsgiga** [comment]
Makes total sense to me @gtmanfred, my bad on that! Will update the PR shortly!

Thank you!

**Comment by rallytime** [comment]
Closed via #46333
